tests/OmniSharp.Roslyn.CSharp.Tests/FormattingFacts.cs

```diff
@@ -128,6 +130,116 @@ public void M ()
                 new LinePositionSpanTextChange { StartLine = 2, StartColumn = 30, EndLine = 3, EndColumn = 0, NewText = "\n " });
         }
 
+        [Fact]
+        public async Task TextChangesOnStaringSpanBeforeFirstCharacterInLine()
+        {
+            var source = new[]
+            {
+                "class Program",
+                "{",
+                "    public static void Main()",
+                "    {",
+                "[|               int foo = 1;|]",
+                "    }",
+                "}",
+            };
+
+            var expected =
+@"class Program
+{
+    public static void Main()
+    {
+        int foo = 1;
+    }
+}";
+
+            await AssertTextChanges(string.Join("\r\n", source), expected);
+        }
+
+        [Fact]
+        public async Task TextChangesOnStartingSpanAtFirstCharacterInLine()
+        {
+            var source = new[]
+            {
+                "class Program",
+                "{",
+                "    public static void Main()",
+                "    {",
+                "               [|int foo = 1;|]",
+                "    }",
+                "}",
+            };
+
+            var expected =
+@"class Program
+{
+    public static void Main()
+    {
+        int foo = 1;
+    }
+}";
+
+            await AssertTextChanges(string.Join("\r\n", source), expected);
+        }
+
+        [Fact]
+        public async Task TextChangesOnStartingSpanAfterFirstCharacterInLine()
+        {
+            var source = new[]
```

Comment:
I think you are storing all of these as string arrays and immediately string.join ing them. Might as well use string literals...